File: src/_2025/Arrays_Hashing/ContainsDuplicate.java
package _2025.Arrays_Hashing;

import java.util.HashSet;

public class ContainsDuplicate {

    private boolean hasDuplicate(int[] nums){
        int originalLength = nums.length;
        HashSet<Integer> uniqueElement = new HashSet<>();
        for( int num : nums) uniqueElement.add(num);
        return uniqueElement.size() != originalLength;
    }

    public void run(String[] args) {
       int[] nums = new int[]{1,2,3,5};
       System.out.println(this.hasDuplicate(nums));
    }
}

File: src/_2026/Python/Arrays/contains_duplicate.py
def contains_duplicate(nums: list[int]):
    # temp_arr = []
    # for num in nums:
    #     if temp_arr.count(num) > 0:
    #         return True
    #     temp_arr.append(num)
    # return False

    hash_set = set(nums)
    return len(hash_set) != len(nums)


print(contains_duplicate([1, 2, 3, 1]))
